xarray/core/dataset.py

```diff
@@ -792,13 +806,19 @@ def chunks(self):
         array.
         """
         chunks = {}
-        for v in self.variables.values():
+        for v in self.data_vars.values():
             if v.chunks is not None:
                 new_chunks = list(zip(v.dims, v.chunks))
                 if any(chunk != chunks[d] for d, chunk in new_chunks
                        if d in chunks):
                     raise ValueError('inconsistent chunks')
                 chunks.update(new_chunks)
+        if chunks:
```

Comment:
Why should this need `chunks` to not be empty already? That seems strange (maybe backwards) to me.

I might simply make this:
```
for dim, size in self.dims.items():
    if dim not in chunks:
        chunks[dim] = (size,)
```